### PR TITLE
jwt-based google openid connect example

### PR DESCRIPTION
**Pull request by washort**
This uses the newer Google login API, sending a JWT to the backend with credentials. This version calls out to google to verify the token for each request, we could either use our own token for the session or do the verification work in-process, but I figured I'd start simple.

Based on docs from here:
https://developers.google.com/identity/sign-in/web/sign-in
https://developers.google.com/identity/sign-in/web/backend-auth

### CODE REVIEW

File: shielddash/api/static/example.html [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<!DOCTYPE html>
+<meta name="google-signin-client_id" content="{{ GOOGLE_AUTH_KEY }}">
```
Comment (spasovski) [inline shielddash/api/static/example.html]: Loving the lack of `<head>` and `<body>` 👍

File: requirements.txt [hunk 2]
```diff
@@ -1,7 +1,19 @@
+dj-database-url==0.4.0
 Django==1.9.5
 djangorestframework==3.3.3
 django-filter==0.13.0
-gunicorn==19.4.5
+google-api-python-client==1.5.0
+ gunicorn==19.4.5
```
Comment (robhudson) [inline requirements.txt:6]: looks like a space here

File: shielddash/settings.py [hunk 3]
```diff
@@ -42,11 +45,16 @@
     'django.contrib.sessions',
     'django.contrib.staticfiles',
     'shielddash.studies',
+
+    'rest_framework',
+    'shielddash.api',
```
Comment (robhudson) [inline shielddash/settings.py]: Don't think this exists anymore.

File: shielddash/settings.py [hunk 4]
```diff
@@ -42,11 +45,16 @@
...
 ]
 
 MIDDLEWARE_CLASSES = [
     'django.middleware.security.SecurityMiddleware',
     'django.middleware.common.CommonMiddleware',
+    'django.contrib.sessions.middleware.SessionMiddleware',
+    'django.contrib.auth.middleware.AuthenticationMiddleware',
     'django.middleware.csrf.CsrfViewMiddleware',
     'django.contrib.sessions.middleware.SessionMiddleware',
     'django.contrib.auth.middleware.AuthenticationMiddleware',
```
Comment (robhudson) [inline shielddash/settings.py:56]: The above 2 lines are duplicated here.

File: shielddash/studies/authentication.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+                                     'https://accounts.google.com']:
+                raise crypt.AppIdentityError("Wrong issuer.")
+            if idinfo['hd'] != settings.GOOGLE_AUTH_HOSTED_DOMAIN:
+                raise crypt.AppIdentityError("Wrong hosted domain.")
+        except crypt.AppIdentityError, e:
+            return exceptions.AuthenticationFailed(e)
+        (u, isCreated) = get_user_model().objects.get_or_create(
+            email=idinfo['email'],
+            first_name=idinfo['given_name'],
+            last_name=idinfo['family_name'])
```
Comment (robhudson) [inline shielddash/studies/authentication.py:44]: Will this just be using the default django auth `User` model?

Comment (washort) [inline shielddash/studies/authentication.py:44]: Yes, but the point of calling `get_user_model` is that it'll work with any replacement we need.

File: shielddash/studies/static/foo.js [hunk 6]
```diff
@@ -0,0 +1 @@
+alert("y'all ready");
```
Comment (robhudson) [inline shielddash/studies/static/foo.js]: Is this file needed?

**Comment by robhudson** [comment]
Some extra tests for the auth stuff would be good here.

r+wc

File: shielddash/studies/static/foo.js (hunk 6 above)
Comment (washort) [inline shielddash/studies/static/foo.js]: i already deleted this from one version of this branch... git is hard
